language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ImplicitNewExpressionNodeContext.java

```diff
@@ -84,47 +83,47 @@ public ImplicitNewExpressionNodeContext() {
                 return Optional.empty();
         }
 
-        Scope.ScopeEntry scopeEntry = null;
-        List<Scope.ScopeEntry> visibleSymbols = new ArrayList<>(context.get(CommonKeys.VISIBLE_SYMBOLS_KEY));
-            if (this.onQualifiedNameIdentifier(context, typeDescriptor)) {
+        Optional<Symbol> nameReferenceSymbol = Optional.empty();
+        List<Symbol> visibleSymbols = new ArrayList<>(context.get(CommonKeys.VISIBLE_SYMBOLS_KEY));
+        if (this.onQualifiedNameIdentifier(context, typeDescriptor)) {
             QualifiedNameReferenceNode nameReferenceNode = (QualifiedNameReferenceNode) typeDescriptor;
 
-            Optional<Scope.ScopeEntry> pkgSymbol = CommonUtil.packageSymbolFromAlias(context,
+            Optional<ModuleSymbol> pkgSymbol = CommonUtil.searchModuleForAlias(context,
                     QNameReferenceUtil.getAlias(nameReferenceNode));
-            if (!pkgSymbol.isPresent()) {
+            if (pkgSymbol.isEmpty()) {
                 return Optional.empty();
             }
-            scopeEntry = ((BPackageSymbol) pkgSymbol.get().symbol).scope.entries.entrySet().stream()
-                    .filter(entry -> entry.getKey().value.equals(nameReferenceNode.identifier().text()))
-                    .map(Map.Entry::getValue)
-                    .findAny()
-                    .orElse(null);
+            nameReferenceSymbol = pkgSymbol.get().allSymbols().stream()
+                    .filter(symbol -> symbol.name().equals(nameReferenceNode.identifier().text()))
+                    .findAny();
```

Comment:
`findAny()` will add some randomness into selection criteria even though we expect only one matching symbol. Shall we use `findFirst()` for the sake of consistent in the event of multiple results.